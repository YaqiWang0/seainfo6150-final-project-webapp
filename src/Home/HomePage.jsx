import React, { useEffect, useState} from "react"
import StockRanking from "./StockRanking/StockRanking";
import NavigationBar from "../NavigationBar/NavigationBar";
import NewsList from "../News/NewsList";
import style from "./HomePage.module.css";
import { useHistory, useLocation } from "react-router-dom";
import Slides from "../Slides/Slides";


const HomePage = (props) => {
    let location = useLocation();

    const [loginCredential,setLoginCredential] = useState({})

    if(location.state !== undefined) {

        if(location.state.login == true) {
            setLoginCredential({
                login: location.state.login,
                userId: location.state.userId,
                email: location.state.email
            })
        } else {
            setLoginCredential({
                login: location.state.login,
                userId: "",
                email: ""
            })
        }
        location.state = undefined
    }

     return (
        <div className={style.background}>
        <div className={style.container}>
            <NavigationBar login = {loginCredential}/>
            <div className={style.picture}>
            <Slides news = {Object.values(props.fetchedData)}/>
            </div>
            <StockRanking gainer = {Object.values(props.gainer)} loser = {Object.values(props.loser)} login = {loginCredential}/>
            <NewsList data = {Object.values(props.fetchedData)} />
        </div>
            </div>
    )
}

export default HomePage