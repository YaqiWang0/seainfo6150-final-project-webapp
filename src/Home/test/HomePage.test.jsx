import React from "react";
import HomePage from "../HomePage";
import data from "../test/data";


describe("Home Page test", () => {
    it("render correctly when login", () => {
        jest.mock('react-router-dom', () => ({
            ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
            useParams: () => ({
                userId: '12345',
                login: true,
                email: 'test@gmail.com'
            }),
            useRouteMatch: () => ({ url: '/' }),
        }));
        const { container  } = render(<HomePage fetchedData = {data.news} gainer = {data.gainer} loser = {data.loser}/>);
        expect(container ).toMatchSnapshot();
    })
    it("render correctly when logout", () => {
        jest.mock('react-router-dom', () => ({
            ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
            useParams: () => ({
                login: false
            }),
            useRouteMatch: () => ({ url: '/' }),
        }));
        const { container  } = render(<HomePage fetchedData = {data.news} gainer = {data.gainer} loser = {data.loser}/>);
        expect(container ).toMatchSnapshot();
    })
})