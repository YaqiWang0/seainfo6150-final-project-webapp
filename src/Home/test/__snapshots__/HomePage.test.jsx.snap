// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page test render correctly when login 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="container"
        >
          <div
            class="pic"
          />
        </div>
        <div
          class="func"
        >
          <div>
            <form
              action=""
              class="searchBar"
            >
              <input
                name="search"
                pattern=".*\\\\S.*"
                required=""
                type="search"
              />
              <button
                class="searchBtn"
                type="submit"
              >
                <span>
                  Search
                </span>
              </button>
            </form>
          </div>
          <div
            class="buttons"
          >
            <div
              class="show"
            >
              <nav>
                <ul>
                  <li>
                    LOGIN
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="hide"
            >
              <nav>
                <ul>
                  <li>
                    USER
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="hide"
            >
              <nav>
                <ul>
                  <li>
                    LOGOUT
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
      <div
        class="picture"
      >
        <div
          class="container"
          style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
        >
          <div
            class="cover"
          >
            <div
              class="left_arrow"
              role="left"
            >
              <a
                class="arrow_left"
                href="#"
              />
            </div>
            <div
              class="right_arrow"
              role="right"
            >
              <a
                class="arrow_right"
                href="#"
              />
            </div>
            <div
              class="text"
            >
              5 Stocks Growing Earnings
            </div>
            <ul
              class="dots"
            >
              <li
                class="dotSelected"
                role="dot"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ranking_container"
      >
        <div
          class="gainer"
        >
          <div
            class="title"
          >
             Most Gainer Stock Today
          </div>
          <div
            class="list"
          >
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SNOA
                  </span>
                  <span
                    class="company_name"
                  >
                    Sonoma Pharmaceuticals Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $10.75
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SLDB
                  </span>
                  <span
                    class="company_name"
                  >
                    Solid Biosciences Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $5.7706
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    EQS
                  </span>
                  <span
                    class="company_name"
                  >
                    EQUUS Total Return Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $1.755
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    NH
                  </span>
                  <span
                    class="company_name"
                  >
                    NantHealth Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $4.18
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    VTVT
                  </span>
                  <span
                    class="company_name"
                  >
                    vTv Therapeutics Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $3.2
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
          </div>
        </div>
        <div
          class="loser"
        >
          <div
            class="title"
          >
             Most Loser Stock Today
          </div>
          <div
            class="list"
          >
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    GLSI
                  </span>
                  <span
                    class="company_name"
                  >
                    Greenwich LifeSciences, Inc.
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $46.52
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SLS
                  </span>
                  <span
                    class="company_name"
                  >
                    Sellas Life Sciences Group Inc
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $11.7
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    NBRV
                  </span>
                  <span
                    class="company_name"
                  >
                    Nabriva Therapeutics PLC
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $2.455
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    KBLMU
                  </span>
                  <span
                    class="company_name"
                  >
                    KBL Merger Corp IV
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $8.38
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    IPDN
                  </span>
                  <span
                    class="company_name"
                  >
                    Professional Diversity Network, Inc.
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $1.495
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="5 Stocks Growing Earnings"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  5 Stocks Growing Earnings
                </div>
                <div
                  class="symbol"
                >
                  NSIT
                </div>
              </div>
              <div
                class="content"
              >
                According to the GuruFocus All-in-One Screener, a Premium feature, as of Dec. 11, the following guru-held companies have positive future earnings estimates from Morningstar analysts.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/m/q/medicine-5103043-1920-27.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate
                </div>
                <div
                  class="symbol"
                >
                  PFE
                </div>
              </div>
              <div
                class="content"
              >
                Pfizer Inc. (NYSE: PFE) and BioNTech SE's – ADR (NASDAQ: BNTX) coronavirus vaccine candidate passed the Adcom test, with the FDA panel voting 17-4 in favor of its risk-benefit profile in individuals 16 and older.  Here's SVB Leerink analyst Geoffrey Porges' take on the dynamics of the Adcom vote.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Oil futures inch lower for the session, but climb for the week"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/5/p/etf21.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Oil futures inch lower for the session, but climb for the week
                </div>
                <div
                  class="symbol"
                >
                  OIL
                </div>
              </div>
              <div
                class="content"
              >
                Oil futures inched lower on Friday, a day after settling at their highest in nine months, but tallied a gain for the week, amid expectations that the approval of a COVID-19 vaccine in the U.S. will provide a lift to the economy, and demand for energy. January West Texas Intermediate crude fell 21 cents, or nearly 2.5%, to settle at $46.57 a barrel on the New York Mercantile Exchange.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="White House threatens to fire FDA chief unless Pfizer Covid vaccine approved Friday, report says"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/7/d/106256485-1574272467663gettyimages-1183636908r.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  White House threatens to fire FDA chief unless Pfizer Covid vaccine approved Friday, report says
                </div>
                <div
                  class="symbol"
                >
                  PFE
                </div>
              </div>
              <div
                class="content"
              />
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/7/w/becton-dickinson-covidjpgquality90stripall.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company
                </div>
                <div
                  class="symbol"
                >
                  BDXA
                </div>
              </div>
              <div
                class="content"
              >
                The health care sector is considered a defensive sector as people will seek out treatment for illness and aliments regardless of the status of the economy. Many companies in this sector literally provide products that consumers cannot live without.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Home Page test render correctly when logout 1`] = `
<div>
  <div
    class="background"
  >
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="container"
        >
          <div
            class="pic"
          />
        </div>
        <div
          class="func"
        >
          <div>
            <form
              action=""
              class="searchBar"
            >
              <input
                name="search"
                pattern=".*\\\\S.*"
                required=""
                type="search"
              />
              <button
                class="searchBtn"
                type="submit"
              >
                <span>
                  Search
                </span>
              </button>
            </form>
          </div>
          <div
            class="buttons"
          >
            <div
              class="show"
            >
              <nav>
                <ul>
                  <li>
                    LOGIN
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="hide"
            >
              <nav>
                <ul>
                  <li>
                    USER
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
            <div
              class="hide"
            >
              <nav>
                <ul>
                  <li>
                    LOGOUT
                    <span />
                    <span />
                    <span />
                    <span />
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
      <div
        class="picture"
      >
        <div
          class="container"
          style="background-image: url(https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg);"
        >
          <div
            class="cover"
          >
            <div
              class="left_arrow"
              role="left"
            >
              <a
                class="arrow_left"
                href="#"
              />
            </div>
            <div
              class="right_arrow"
              role="right"
            >
              <a
                class="arrow_right"
                href="#"
              />
            </div>
            <div
              class="text"
            >
              5 Stocks Growing Earnings
            </div>
            <ul
              class="dots"
            >
              <li
                class="dotSelected"
                role="dot"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
              <li
                class="dotUnSelected"
              />
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ranking_container"
      >
        <div
          class="gainer"
        >
          <div
            class="title"
          >
             Most Gainer Stock Today
          </div>
          <div
            class="list"
          >
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SNOA
                  </span>
                  <span
                    class="company_name"
                  >
                    Sonoma Pharmaceuticals Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $10.75
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SLDB
                  </span>
                  <span
                    class="company_name"
                  >
                    Solid Biosciences Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $5.7706
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    EQS
                  </span>
                  <span
                    class="company_name"
                  >
                    EQUUS Total Return Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $1.755
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    NH
                  </span>
                  <span
                    class="company_name"
                  >
                    NantHealth Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $4.18
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    VTVT
                  </span>
                  <span
                    class="company_name"
                  >
                    vTv Therapeutics Inc
                  </span>
                </div>
                <span
                  class="gainer_price_text"
                >
                  $3.2
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
          </div>
        </div>
        <div
          class="loser"
        >
          <div
            class="title"
          >
             Most Loser Stock Today
          </div>
          <div
            class="list"
          >
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    GLSI
                  </span>
                  <span
                    class="company_name"
                  >
                    Greenwich LifeSciences, Inc.
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $46.52
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    SLS
                  </span>
                  <span
                    class="company_name"
                  >
                    Sellas Life Sciences Group Inc
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $11.7
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    NBRV
                  </span>
                  <span
                    class="company_name"
                  >
                    Nabriva Therapeutics PLC
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $2.455
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    KBLMU
                  </span>
                  <span
                    class="company_name"
                  >
                    KBL Merger Corp IV
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $8.38
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
            <div>
              <div
                class="ranking_item"
                role="click"
              >
                <div
                  class="name"
                >
                  <span
                    class="stock_name"
                  >
                    IPDN
                  </span>
                  <span
                    class="company_name"
                  >
                    Professional Diversity Network, Inc.
                  </span>
                </div>
                <span
                  class="loser_price_text"
                >
                  $1.495
                </span>
              </div>
              <hr
                class="line"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="5 Stocks Growing Earnings"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/h/h/bussines-services9.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  5 Stocks Growing Earnings
                </div>
                <div
                  class="symbol"
                >
                  NSIT
                </div>
              </div>
              <div
                class="content"
              >
                According to the GuruFocus All-in-One Screener, a Premium feature, as of Dec. 11, the following guru-held companies have positive future earnings estimates from Morningstar analysts.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/m/q/medicine-5103043-1920-27.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Pfizer Analyst Breaks Down The FDA Process For Coronavirus Vaccine Candidate
                </div>
                <div
                  class="symbol"
                >
                  PFE
                </div>
              </div>
              <div
                class="content"
              >
                Pfizer Inc. (NYSE: PFE) and BioNTech SE's – ADR (NASDAQ: BNTX) coronavirus vaccine candidate passed the Adcom test, with the FDA panel voting 17-4 in favor of its risk-benefit profile in individuals 16 and older.  Here's SVB Leerink analyst Geoffrey Porges' take on the dynamics of the Adcom vote.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Oil futures inch lower for the session, but climb for the week"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/5/p/etf21.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Oil futures inch lower for the session, but climb for the week
                </div>
                <div
                  class="symbol"
                >
                  OIL
                </div>
              </div>
              <div
                class="content"
              >
                Oil futures inched lower on Friday, a day after settling at their highest in nine months, but tallied a gain for the week, amid expectations that the approval of a COVID-19 vaccine in the U.S. will provide a lift to the economy, and demand for energy. January West Texas Intermediate crude fell 21 cents, or nearly 2.5%, to settle at $46.57 a barrel on the New York Mercantile Exchange.
              </div>
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="White House threatens to fire FDA chief unless Pfizer Covid vaccine approved Friday, report says"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/7/d/106256485-1574272467663gettyimages-1183636908r.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  White House threatens to fire FDA chief unless Pfizer Covid vaccine approved Friday, report says
                </div>
                <div
                  class="symbol"
                >
                  PFE
                </div>
              </div>
              <div
                class="content"
              />
            </div>
          </div>
        </div>
        <div>
          <hr
            class="line"
          />
          <div
            class="container"
          >
            <img
              alt="Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company"
              class="pic"
              src="https://cdn.snapi.dev/images/v1/7/w/becton-dickinson-covidjpgquality90stripall.jpg"
            />
            <div
              class="contentBlock"
            >
              <div
                class="titleBlock"
              >
                <div
                  class="title"
                >
                  Becton, Dickinson & Co. Is a Highly Profitable, Recession-Proof Company
                </div>
                <div
                  class="symbol"
                >
                  BDXA
                </div>
              </div>
              <div
                class="content"
              >
                The health care sector is considered a defensive sector as people will seek out treatment for illness and aliments regardless of the status of the economy. Many companies in this sector literally provide products that consumers cannot live without.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
